### PR TITLE
feat: enable re-run OCR

### PR DESCRIPTION
**Pull request by alex-krasn**

### CODE REVIEW

File: src/react/components/pages/editorPage/canvas.scss [hunk 1]
```diff
@@ -86,3 +86,8 @@
   width: 100%;
   height: 100%;
 }
+
+.re-run-ocr-item {
+margin-left: 0.5rem;
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/canvas.scss:91]: indent

File: src/react/components/pages/editorPage/canvas.tsx [hunk 2]
```diff
@@ -938,14 +952,19 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         });
     }
 
+    private setRerunOCR = async () => {
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/canvas.tsx:955]: handler should handle functionality, not componentDidUpdate

File: src/react/components/pages/editorPage/canvas.tsx [hunk 3]
```diff
@@ -938,14 +946,19 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         });
     }
 
-    private loadOcr = async () => {
+    private rerunOcr = () => {
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/canvas.tsx:949]: I don't this this wrapper function is necessary, you could just pass loadOcr as prop to command bar right?

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 4]
```diff
@@ -629,37 +629,38 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
...
-                            } catch (err) {
-                                this.updateAssetState(asset.id, false);
-                                this.setState({
-                                    isError: true,
-                                    errorTitle: err.title,
-                                    errorMessage: err.message,
-                                });
-                            }
+                        if (asset && (asset.state === AssetState.NotVisited || rerunOCR)) {
+                                try {
```
Comment (stew-ro) [inline src/react/components/pages/editorPage/editorPage.tsx:650]: indented too much

**Comment by kunzms** [comment]
Yes, it makes sense to use the same terminology, I prefer "document" as customers may be more familiar with "document".

**Comment by stew-ro** [comment]
> Yes, it makes sense to use the same terminology, I prefer "document" as customers may be more familiar with "document".

"Document" sounds good.

So, should the document preview button be renamed "Run OCR on all unvisited documents" and the command bar buttons renamed to "Run OCR on all documents?" I don't think "Rerun" makes sense because the user may not have already ran OCR
image

**Comment by alex-krasn** [comment]
This closing issue #256

**Pull request merged**